### ISSUE TITLE
&Errno::* issue 

### ISSUE THREAD

**Issue by atoomic** [CpanelInc/perl-compiler#36]
Undefined subroutine ... when compiling this code
```
use Errno; print qq{ok\n} if &Errno::EBADF
```

This is not happening when using a function or method call directly
```
use Errno; print qq{ok\n} if Errno::EBADF();
use Errno; print qq{ok\n} if Errno->EBADF();
```

Note that Errno is a tied module.
At compile time the RV->IV is converted to a GV, which is the same behavior with an uncompiled perl, but the function cannot be retrieved correctly....

Monkey patch
```diff
diff --git a/lib/B/C/OverLoad/B/GV.pm b/lib/B/C/OverLoad/B/GV.pm
index b47d25a15..f17b9715e 100644
--- a/lib/B/C/OverLoad/B/GV.pm
+++ b/lib/B/C/OverLoad/B/GV.pm
@@ -43,14 +43,24 @@ my $CORE_SYMS = {
 our $under = '';
 our @under = ();

-
 sub do_save {
     my ( $gv, $name ) = @_;

     # return earlier for special cases
     return $CORE_SYMS->{ $gv->get_fullname } if $gv->is_coresym();

-    # This is probably dead code.
+    # workardound for Errno constant values
+    if ( $gv->get_fullname =~ qr{^Errno::(.+)$} ) {
+        my $errno_const = $1;
+        if ( my $sub = 'Errno'->can($errno_const) ) {
+            my $const = eval qq{\&Errno::${errno_const}};
+            if ( $const =~ qr{^[0-9]+$} ) {
+                $const = int($const);
+                my $v = eval qq[sub { $const }];
+                return B::svref_2object( \$v )->save( $gv->get_fullname );
+            }
+        }
+    }

     # STATIC_HV need to handle $0
```

**Comment by atoomic** [comment]
need to revert commit aa21a80ff13863ae8dfe470d8d816968303c36f6 once fix
```
commit aa21a80ff13863ae8dfe470d8d816968303c36f6
Author: Nicolas Rochelemagne <rochelemagne@cpanel.net>
Date:   Tue Jun 27 09:57:39 2017 -0600

    Adjust tests for &Errno::* calls

    Several tests are failing due to a call
    to Errno function via a reference.

    This commit should be reverted once the real
    Errno fix is done.
```

**Comment by atoomic** [comment]
seems like a problem while saving const_sv_xsub, we need to set
```
         CvXSUBANY(cv).any_ptr = const_sv;
         CvXSUB(cv) = const_sv_xsub;
```

**Comment by atoomic** [comment]
closed with commit a81b0cb which is also fixing the upstream issue #305
